src/main/java/org/apache/commons/lang3/ObjectUtils.java

```diff
@@ -638,7 +642,7 @@ public static String toString(final Object obj, final String nullStr) {
      * @return a negative value if c1 &lt; c2, zero if c1 = c2
      *  and a positive value if c1 &gt; c2
      */
-    public static <T extends Comparable<? super T>> int compare(final T c1, final T c2) {
+    public static <T extends Comparable<? super T>> int compare(final @PolySameLen @PolyLowerBound @PolyUpperBound T c1, final @PolySameLen @PolyLowerBound @PolyUpperBound T c2) {
```

Comment:
What is the purpose of these annotations? They don't make sense to me.

Reply:
Because checker issues an error on `c1.compareTo(c2)` saying that these annotations are required.